File: 0071-simplify-path/0071-simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        stack<string> st;
        st.push("/");
        string tempt = "";
        bool newdir = false;
        bool dots = true;
        for(int i = 0; i< path.size();i++){
            if(path[i] != '/') tempt += path[i];
            
            else{
               
                if(tempt == ".."){
                    
                    if (st.size() > 1) {
                        
                        while(st.top() == "/" and st.size()>1) st.pop();
                         if (st.size()>1) st.pop();
                    }                    
                }
                
                else if(tempt == ".")  tempt = "";
                
                else if((st.top() == "/")  and tempt == "") continue;
                
                else if (tempt!= "") st.push(tempt);
                
                tempt = "";
                
                if (st.top() != "/" and i != path.size() - 1) st.push("/");
                
            }    
        }
        
       if(tempt == ".."){
                    
                    if (st.size() > 1) {
                        
                        while(st.top() == "/" and st.size()>1) st.pop();
                         if (st.size()>1) st.pop();
                    }                    
                }
                
                else if(tempt == ".")  tempt = "";
                
                else if((st.top() == "/")  and tempt == "");
                
                else if (tempt!= "") st.push(tempt);
                
                
         

        string result = "";
        vector<string> temp;
      
        while(st.size() > 0){
            
            temp.push_back( st.top());
            st.pop();
        }
        
        reverse(temp.begin(), temp.end());
        
        for(int i = 0; i < temp.size(); i++){
            
            if((i == (temp.size() - 1)) and temp[i] == "/") continue;
            result += temp[i];
        }
        
        if (result == "") return "/";
        return result;
        
        
    }
};

File: 0071-simplify-path/0071-simplify-path.go
func simplifyPath(path string) string {
    st := make([]string,1);
    tempt := ""
    
    for i, _ := range path{
        
        if path[i] != '/' {
            tempt += string(rune(path[i]))
        } else{
               
            if tempt == ".."{
                    
                if len(st) > 1 {
                        
                    for  st[len(st)-1] == "/" && len(st) > 1{
                        st = st[:len(st)-1]
                    }
                    if len(st) >1 {
                     st = st[:len(st)-1]
                    }
                }  
                
            }else if tempt == "."{
                tempt = ""
            }  else if (st[len(st)-1] == "/")  && tempt == ""{
                continue
            } else if (tempt!= "") {
                st = append(st,tempt)
            }
                
            tempt = ""
                
            if st[len(st)-1] != "/" && i != len(path) - 1 {
                st = append(st,"/")
            }
                
         }  
    }
    
    if tempt == ".."{
                    
        if (len(st) > 1) {
                        
            for st[len(st)-1] == "/" && len(st) > 1{
                st = st[:len(st)-1]
            }
            if (len(st) > 1) {
                st = st[:len(st)-1]
            }
                    }                    
    } else if tempt == "."{
        tempt = ""
    } else if (st[len(st)-1] == "/")  && tempt == ""{
        
    }else if tempt!= ""{
        st = append(st,tempt)
    }
                
                
        
    result := "";
    temp := make([]string,0)
      
    for len(st) > 0{
            
        temp = append(temp,st[len(st)-1])
        st = st[:len(st)-1]
     }
    
    for i, j := 0, len(temp)-1; i < j; i, j = i+1, j-1 {
        temp[i], temp[j] = temp[j], temp[i]
    }
        
        
    for i := 0; i < len(temp); i++{
            
        if(i == (len(temp) - 1)) && temp[i] == "/" {
            continue
        }
        result += temp[i]
    }
        
    if result == ""{
        return "/"
    } 
    return result
    
}